File: src/main/java/guru/xgm/language/angelscript/psi/AngelScriptElementFactory.java
package guru.xgm.language.angelscript.psi;

import com.intellij.openapi.project.Project;
import com.intellij.psi.PsiElement;
import com.intellij.psi.PsiFileFactory;
import guru.xgm.language.angelscript.extapi.psi.AngelScriptPsiFileBase;
import guru.xgm.language.angelscript.lang.AngelScriptLanguage;
import org.jetbrains.annotations.NotNull;

public class AngelScriptElementFactory {

    private static AngelScriptPsiFileBase createFile(Project project, String text) {
        String name = "dummy.ass";
        return (AngelScriptPsiFileBase) PsiFileFactory.getInstance(project).createFileFromText(name, AngelScriptLanguage.INSTANCE, text);
    }

    public static AngelScriptExpr recreateExpr(Project project, String value) {
        final var file = createFile(project, "int fuckingCrutch = " + value + ";");
        return ((AngelScriptVar) file.getFirstChild()).getExprList().get(0);
    }

    public static void replaceExprChild(@NotNull Project project, @NotNull PsiElement target, @NotNull String value) {
        final var child = recreateExpr(project, value).getFirstChild();
        if (child == null) return;
        target.replace(child);
    }
}

File: src/main/kotlin/guru/xgm/language/angelscript/psi/AngelScriptElementFactory.kt
package guru.xgm.language.angelscript.psi

import com.intellij.openapi.project.Project
import com.intellij.psi.PsiElement
import com.intellij.psi.PsiFileFactory
import guru.xgm.language.angelscript.extapi.psi.AngelScriptPsiFileBase
import guru.xgm.language.angelscript.lang.AngelScriptLanguage.Companion.instance

object AngelScriptElementFactory {
    private fun createFile(project: Project, text: String): AngelScriptPsiFileBase {
        val name = "dummy.as"
        return PsiFileFactory.getInstance(project).createFileFromText(name, instance, text) as AngelScriptPsiFileBase
    }

    private fun recreateExpr(project: Project, value: String): AngelScriptExpr {
        val file = createFile(project, "int fuckingCrutch = $value;")
        return (file.firstChild as AngelScriptVar).exprList[0]
    }

    fun replaceExprChild(project: Project, target: PsiElement, value: String) {
        val child = recreateExpr(project, value).firstChild ?: return
        target.replace(child)
    }
}
